portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,27 +40,42 @@
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    String defaultCommentLimit = "5";
+    int commentLimit = Integer.parseInt(getParameter(request, "numShown", defaultCommentLimit));
```

Comment:
What happens if I call /data?numShown=lol
or /data?numShown=
or /data?numShown=99999999999999999999999999999999999999

Reply:
Now using NumberFormatException to make sure it is parsed as a valid int. Will add an upper bound feature in a future commit

Reply:
I'm using a similar approach in the JS portion of things.

Reply:
New JS code to prevent malformed inputs has been added by substituting them for a default value; their is also some additional Java code to handle malformed inputs in place.